packages/core/src/compiler.ts

```diff
@@ -182,16 +195,20 @@ export default class Compiler {
           }
         }
 
-        args.push(JSON.stringify(this.plural.locale));
+        if (!isLocaleSpecificFormattingModule) {
+          args.push(JSON.stringify(this.plural.locale));
+        }
         if (token.param) {
           if (pluralToken && this.options.strict) pluralToken = null;
           const arg = this.getFormatterArg(token, pluralToken);
           if (arg) args.push(arg);
         }
-        fn = token.key;
-        this.setFormatter(fn);
+        fn = isLocaleSpecificFormattingModule
+          ? `${token.key}__${this.currentLocaleCode?.replace(/[^a-zA-Z]/g, '')}`
```

Comment:
e.g. `es-US` => `esUS`

Reply:
Rather than putting together a new sanitization solution, it'd be better to use [`identifier()`](https://www.npmjs.com/package/safe-identifier#identifierkey-string-unique-boolean-string).